cairo/ethereum/cancun/keth/teardown.cairo

```diff
@@ -90,160 +90,173 @@ func teardown{
-
-    assert [output_ptr] = teardown_commitment.value.low;
-    assert [output_ptr + 1] = teardown_commitment.value.high;
-    let output_ptr = output_ptr + 2;
-
-    // Commit to the same inputs as the body.cairo's outputs.
-    let body_commitment = body_commitments{
-        range_check_ptr=range_check_ptr,
-        bitwise_ptr=bitwise_ptr,
-        keccak_ptr=keccak_ptr,
-        poseidon_ptr=poseidon_ptr,
-    }(
-        header_commitment,
-        block_transactions,
-        state,
-        transactions_trie,
-        receipts_trie,
-        block_logs,
-        block_hashes,
-        gas_available,
-        chain_id,
-        excess_blob_gas,
-    );
-    assert [output_ptr] = body_commitment.value.low;
-    assert [output_ptr + 1] = body_commitment.value.high;
-    let output_ptr = output_ptr + 2;
-
-    // Commit to the same inputs as the body.cairo's outputs.
-
-    // Start the teardown process when _apply_body_inner stops.
-    tempvar block_gas_used = Uint(block.value.header.value.gas_limit.value - gas_available.value);
-    let block_logs_bloom = logs_bloom(block_logs);
-
-    _process_withdrawals_inner{state=state, trie=withdrawals_trie}(0, block.value.withdrawals);
-
-    // Squash the receipts, transactions, and withdrawals dicts once they're no longer being modified.
-    default_dict_finalize(
-        cast(transactions_trie.value._data.value.dict_ptr_start, DictAccess*),
-        cast(transactions_trie.value._data.value.dict_ptr, DictAccess*),
-        0,
-    );
-    default_dict_finalize(
-        cast(receipts_trie.value._data.value.dict_ptr_start, DictAccess*),
-        cast(receipts_trie.value._data.value.dict_ptr, DictAccess*),
-        0,
-    );
-    default_dict_finalize(
-        cast(withdrawals_trie.value._data.value.dict_ptr_start, DictAccess*),
-        cast(withdrawals_trie.value._data.value.dict_ptr, DictAccess*),
-        0,
-    );
-
-    let null_account_roots = OptionalMappingAddressBytes32(cast(0, MappingAddressBytes32Struct*));
-    let tx_trie_typed = EthereumTriesImpl.from_transaction_trie(transactions_trie);
-    let transactions_root = root(tx_trie_typed, null_account_roots, 'keccak256');
-
-    let receipt_trie_typed = EthereumTriesImpl.from_receipt_trie(receipts_trie);
-    let receipts_root = root(receipt_trie_typed, null_account_roots, 'keccak256');
-
-    let withdrawal_trie_typed = EthereumTriesImpl.from_withdrawal_trie(withdrawals_trie);
-    let withdrawals_root = root(withdrawal_trie_typed, null_account_roots, 'keccak256');
-
-    finalize_state{state=state}();
-
-    // rebind state
-    tempvar chain = BlockChain(
-        new BlockChainStruct(blocks=chain.value.blocks, state=state, chain_id=chain.value.chain_id)
-    );
-
-    with_attr error_message("InvalidBlock") {
-        assert block_gas_used = block.value.header.value.gas_used;
-
-        let transactions_root_equal = Bytes32__eq__(
-            transactions_root, block.value.header.value.transactions_root
+    with keccak_ptr {
```

Comment:
why do you prefer a with_keccak_ptr rather than injecting it as a implicit argument directly in the function call?

Reply:
because it's used in all functions, and i don't want to manually add it to all